version/info.go

```diff
@@ -11,41 +11,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package collector
+package version
 
 import (
+	"runtime"
+
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+// Build information. Populated at build-time.
 var (
-	Version  string
-	Revision string
-	Branch   string
+	Version   string
+	Revision  string
+	Branch    string
+	BuildUser string
+	BuildDate string
+	GoVersion = runtime.Version()
 )
 
-type versionCollector struct {
-	metric *prometheus.GaugeVec
+// Map provides the iterable version information.
+var Map = map[string]string{
+	"version":   Version,
+	"revision":  Revision,
+	"branch":    Branch,
+	"buildUser": BuildUser,
+	"buildDate": BuildDate,
+	"goVersion": GoVersion,
 }
 
 func init() {
-	Factories["version"] = NewVersionCollector
```

Comment:
I guess it makes sense to always export the version without making it a collector, but the `version` collector needs to be removed from the list of default collectors in node_exporter.go.